src/Players/AIPlayer.hs

```diff
@@ -87,9 +88,29 @@ getMoveWithMaxScore gameTree = ChessMove ('z', -1) ('z', -1)
 -- The higher the score, the more advantageous the scenario is for the colour
 -- First version: score is the sum of the scores for each piece on the board of the colour minus the scores for each piece of the other colour on the board
 -- Scores per piece: https://cdn-media-1.freecodecamp.org/images/1*e4p9BrCzJUdlqx7KVGW9aA.png
--- TODO: implement + test this [Yiyi]
 score :: ChessBoard -> ChessPieceColour -> Integer
-score chessBoard forColour = -1
+score (ChessBoard pieces _) colour = scoreOfCurrColour - scoreOfOppoColour
+    where currColour = filter (\ (position, piece) -> (getPieceColour piece) == colour) pieces
+          oppoColour = filter (\ (position, piece) -> (getPieceColour piece) /= colour) pieces
+          scoreOfCurrColour = foldr (+) 0 (map (\ (position, piece) -> (pieceToScore piece)) currColour)
+          scoreOfOppoColour = foldr (+) 0 (map (\ (position, piece) -> (pieceToScore piece)) oppoColour) 
+
+
+-- Returns the score of the given piece
+pieceToScore :: ChessPiece -> Integer
+pieceToScore piece
+    | piece == King White   = 900
+    | piece == Queen White  = 90
+    | piece == Rook White   = 50
+    | piece == Bishop White = 30
+    | piece == Knight White = 30
+    | piece == Pawn White   = 10
+    | piece == King Black   = -900    
+    | piece == Queen Black  = -90
+    | piece == Rook Black   = -50
+    | piece == Bishop Black = -30
+    | piece == Knight Black = -30
+    | piece == Pawn Black   = -10
```

Comment:
Since you're already subtracting the score of the opposite colour in the `score` function, then all scores here should be positive because you don't know if the currColour is black or white

```suggestion
-- Returns the score of the given piece
pieceToScore :: ChessPiece -> Integer
pieceToScore (King _)   = 900
pieceToScore (Queen _)  = 90
pieceToScore (Rook _)   = 50
pieceToScore (Bishop _) = 30
pieceToScore (Knight _) = 30
pieceToScore (Pawn _)   = 10
```